backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/LessonsService.cs

```diff
@@ -0,0 +1,44 @@
+﻿using AutoMapper;
+using EasySpeak.Core.BLL.Interfaces;
+using EasySpeak.Core.Common.DTO.Lesson;
+using EasySpeak.Core.DAL.Context;
+using EasySpeak.Core.DAL.Entities;
+using Microsoft.EntityFrameworkCore;
+
+namespace EasySpeak.Core.BLL.Services;
+
+public class LessonsService : BaseService, ILessonsService
+{
+    public LessonsService(EasySpeakCoreContext context, IMapper mapper) : base(context, mapper) { }
+
+    public async Task<ICollection<LessonDto>> GetAllLessonsAsync(FiltersRequest filtersRequest)
+    {
+
+        var lessonsFromContext = _context.Lessons
+            .Include(l => l.Tags)
+            .Include(l => l.Questions).ThenInclude(t => t.Subquestions)
+            .Include(l => l.Subscribers)
```

Comment:
it would be better to do subquery for this

Reply:
I didn't get what you mean, pls clarify

Reply:
It is not a good practice to include all subscribers to calculate their sound
You need to add subquery only for count calculating

Reply:
fixed